src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -594,6 +612,42 @@ private async void ReconnectLoop()
         }
     }
 
+    private async Task PublishEventsAsync()
+    {
+        try
+        {
+            while (!_disposedCancellationTokenSource.IsCancellationRequested)
+            {
+                var hasData = await _eventChannel.Reader.WaitToReadAsync(_disposedCancellationTokenSource.Token).ConfigureAwait(false);
+                if (hasData && _eventChannel.Reader.TryRead(out var eventArgs))
```

Comment:
Apologies for not noticing earlier; this probably could be a `while` to limit thrashing on `WaitToReadAsync` calls.

Reply:
It's fixed.